### DISCUSSION TITLE
[SOLVED] Regressions of new version of "vergilius" style

### DISCUSSION THREAD

**Discussion by sakkamade** [martinrotter/rssguard#502] [Q&A] [answered]
Certain time ago, after some (pretty recent) commits, the article viewer obtained fresh `.css` style update.

While I don't mind such beautification, it introduced a lots of bugs to me, for example with tables, with articles which are having some bits of their own styling, and I was once again left being unable to change the font size there, too (which could be adjusted under "Article browser font" in Settings).

It was introduced with 63865ec539a70189bbe1da89e3b1c90eae3eb61a and 6e10be3cdc1a62893602f6bfc1f796e3257e8cc4 I understand.

Myself, I have simply copied the previous version of "vergilius" (0.24), and keep using it as it looks and works much better in most cases.

My suggestion, however, is that to return that previous version of "vergilius" back with the name of "vergilius-old", or something, and keep it as a separate theme, unless it will be too bothersome to maintain so many styles.

**Comment by martinrotter** [comment, marked as answer]
I see. I did some testing with the style and it worked. I agree that CSS style is rather problematic part of the app, because contents of articles are very diverse and often (semi)-broken.

I will revert `vergilius` to previous version and will introduce new skin instead with different name. Also, as you probably are familiar with possibility of having your own skins.

**Reply by sakkamade** [reply]
> Also, as you probably are familiar with possibility of having your own skins.

Yes, that is how I use old vergilius now.

**Reply by martinrotter** [reply]
OK, skins were reverted. Will release new version today.

**Reply by sakkamade** [reply]
> Also, as you probably are familiar with possibility of having your own skins.

But if you mean creating my own, it is rather problematic for me, because as I try to review that 46kb, 46000+ characters long, file, it sends my CPU fly quite easily, say nothing of it being a one-liner (the `<style>`).

Could there be added some bare `.html` wrapper file, as a plain example?

Or most of that required for it to work properly?

**Reply by martinrotter** [reply]
:D Sure, I will create new skin and make it easier to open in dumb text editors.

For example Visual Studio Code opens those files with long one-liners just fine, btw.

Basically, you can remove the huge `style` element and you will end up pretty much with no formatting:
![image](https://user-images.githubusercontent.com/1255302/134118291-f73112db-986c-4529-9a4c-63f2a70c5cc0.png)

```html
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html lang="en">
  <head>

  <style>
  </style>

  <style>
    html, body, div, span, applet, object, iframe,
    h1, h2, h3, h4, h5, h6, p, blockquote, pre,
    a, abbr, acronym, address, big, cite, code,
    del, dfn, em, img, ins, kbd, q, s, samp,
    small, strike, strong, sub, sup, tt, var,
    b, u, i, center,
    dl, dt, dd, ol, ul, li,
    fieldset, form, label, legend,
    table, caption, tbody, tfoot, thead, tr, th, td,
    article, aside, canvas, details, embed, 
    figure, figcaption, footer, header, hgroup, 
    menu, nav, output, ruby, section, summary,
    time, mark, audio, video {
      font-size: 100%;
      font: inherit;
    }
  </style>

  <style>
    html {
      height: 100%;
    }
    
    body {
      height: 100%;
    }
  </style>
  
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <!-- PERCENTAGE-1 gets replaced by actual title of the message by RSS Guard. -->
  <title>%1</title>
  </head>
  <body>
    %2
  </body>
</html>
```

You need to pay special attention to `%N` places as they get replaced by the actual content in runtime. In above example, `%2` gets replaced by layout for single message specified in `html_single_message.html` file, so when fully "assembled" the HTML for single article view looks similar to below code. No need to say you have to have some knowledge of HTML/CSS styling to get the job done.

```html
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html lang="en">
  <head>

  <style>
  </style>

  <style>
    html, body, div, span, applet, object, iframe,
    h1, h2, h3, h4, h5, h6, p, blockquote, pre,
    a, abbr, acronym, address, big, cite, code,
    del, dfn, em, img, ins, kbd, q, s, samp,
    small, strike, strong, sub, sup, tt, var,
    b, u, i, center,
    dl, dt, dd, ol, ul, li,
    fieldset, form, label, legend,
    table, caption, tbody, tfoot, thead, tr, th, td,
    article, aside, canvas, details, embed, 
    figure, figcaption, footer, header, hgroup, 
    menu, nav, output, ruby, section, summary,
    time, mark, audio, video {
      font-size: 100%;
      font: inherit;
    }
  </style>

  <style>
    html {
      height: 100%;
    }
    
    body {
      height: 100%;
    }
  </style>
  
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <!-- PERCENTAGE-1 gets replaced by actual title of the message by RSS Guard. -->
  <title>SOME TITLE</title>
  </head>
  <body>
    <div class="card filled fluid" dir="auto">
      <h5 class="doc section">SOME TITLE <a class="button small primary" href="%3">🔗URL</a> %6</h5>
      <div class="doc section">
        <div style="text-align: center;">
          %2, %5
        </div>
        <div style="padding: 2px; margin: 0px;">
          %7
        </div>
        %4
      </div>
    </div>
  </body>
</html>
```

**Reply by sakkamade** [reply]
> Basically, you can remove the huge `style` element

I lose nothing of importance by doing it?

There are no such places:

> You need to pay special attention to `%N` places

Or similar?

**Reply by sakkamade** [reply]
> For example Visual Studio Code opens those files with long one-liners just fine

That won't work, you know. 
I like kate, though would love to change it! 
Yet, cannot find anything better...

**Reply by sakkamade** [reply]
> I try to review that 46kb, 46000+ characters long

Ha! the problem is actually that it _is_ a one-liner.

Kate can't properly handle lines of 10000+ characters.

The message as follows:——
> The file ... was opened and contained lines longer than the configured Line Length Limit (10,000 characters).
The longest of those lines was 46,620 characters long

I can temporary raise the limit, of course.

**Comment by sakkamade** [comment]
> Basically, you can remove the huge `style` element

I lose nothing of importance by doing it?

There are no such places:

> You need to pay special attention to `%N` places

Or similar?

**Reply by martinrotter** [reply]
Wait, will write up "empty" skin, you can work with with commentary.

**Reply by martinrotter** [reply]
OK, here you go.

https://github.com/martinrotter/rssguard/tree/master/resources/skins/plain

Just clone to your user data folder. I added some commentary into `html_wrapper.html` file.

**Reply by sakkamade** [reply]
Thank you, great commentaries, too!

**Comment by sakkamade** [comment]
I was wondering, why does the Qt stylesheet file bear the extension of `.css`?

Isn't it awfully confusing? Why not `.qss`?

**Reply by martinrotter** [reply]
Why `.qss`? Nevermind brother, it is even written in documentation.

**Reply by martinrotter** [reply]
Anyway, if you create your own custom skin, mind sharing screenshot when it's done?

**Reply by sakkamade** [reply]
I rather open a PR, if this should be the case. 

[Qt-styling](https://github.com/martinrotter/rssguard/discussions/470) is still only in dreams, by the way.

**Comment by sakkamade** [comment]
Well, I decided to give it a try, since I am familiar with it, still WIP though. 
And since you asked

![image](https://user-images.githubusercontent.com/62730422/135218818-e7612c6a-35b6-4af7-89b6-6211565a9ee4.png)

I am totally happy with such plain style, as it looks really natural;
But I am not sure if this would be worth a PR (or a separate theme), so tell me if it would.

"Published on" won't be translated, of course, so I guess I'd need to change that part.

P.S. I really miss **S**CSS...

**Reply by martinrotter** [reply]
Looks cute. I perfonally find "Link" and "Published on" redundant, but other than that it looks fine.

Thing is that these things are heavily user-dependent. Everyone like something else.

Feel free to create PR with the theme, choose the name as you please. I will merge.

**Reply by sakkamade** [reply]
> I personally find "Link" and "Published on" redundant, but other than that it looks fine.

Right... I am still not sure what should I do with the header...
